### DISCUSSION TITLE
XDrip crashes while retrieving data (from L2 and OOP2)

### DISCUSSION THREAD

**Discussion by Veni4** [NightscoutFoundation/xDrip#2132] [Q&A]
This has been an endless string of problems with the app since last week, when I accidentally accepted the notification that said 'External Calibration: Oop2'. First the app stopped updating, and, a bit later, it started to claim 'Bad sensor data' when getting data through NFC. After restarting my phone, XDrip plain started crashing whenever I tried to do NFC, and a while later, crashing each 20 minutes as it tried to connect to the sensor. After verifying all settings and trying countless times, I did the nuclear option and reinstalled the app. It instantly connected again, not bad, but now, it doesn't crash with NFC, it only continues to crash while trying to retrieve data from the sensor via Bluetooth. 
I hypothesise it has to do with data not going through OOP2 and straight to the app, breaking it somehow, but I don't have anyway to prove it or to fix it. 
Please help then.

**Comment by tzachi-dar** [comment]
How did things work before you accepted the notification?
  Do you have oop2 installed on your phone?
  Do you have the patched app installed on your phone?

What is the value of  "out of process libre algorithm"? Is it set on your phone.

Can you connect your phone to a pc and run there:
adb logcat > file
and send us the file?

**Reply by Veni4** [reply]
XDrip worked fine just before. I had OOP2 installed. XDrip was updated as far as I'm aware.
If you're talking about the option in the app about using the oop algorithm, that option is unchecked, as it is supposed to.
I'll get that file when I can.
Thanks

**Comment by Veni4** [comment]
Update: Reinstalling the app seems to have solved most of the issues, but the app still crashes regularly.

**Reply by tolot27** [reply]
Are there any error messages logged?
Can you provide a `adb logcat` output?

**Reply by Veni4** [reply]
As I answered to another user, I'll get it later...
No error messages, just plain crashing, 'XDrip is not responding' and when opening the app it just goes blank. (I have to manually restart the app for it to work again)
